pulsar-websocket/src/main/java/com/yahoo/pulsar/websocket/AbstractWebSocketHandler.java

```diff
@@ -73,11 +74,16 @@ public void onWebSocketConnect(Session session) {
             }
         }
 
-        if (service.isAuthorizationEnabled() && !isAuthorized(authRole)) {
-            log.warn("[{}] WebSocket Client [{}] is not authorized on topic {}", session.getRemoteAddress(), authRole,
-                    topic);
-            close(WebSocketError.NotAuthorizedError);
-            return;
+        if (service.isAuthorizationEnabled()) {
+            final String role = authRole;
+            isAuthorized(authRole).thenApply(isAuthorized -> {
```

Comment:
This would be kind of different, because when the `onWebSocketConnect()` method completes we should have already done the authorization. 

I think we need to block on the `isAuhtorized()` future, or find a way to have an async `onWebSocketConnect()`